connection.py

```
@@ -0,0 +1,218 @@
+        self.stream_active = True
+        print("Flag is set to True")
+        self.num_channels = num_channels
+
+    def setup_csv(self):
+        if not self.csv_logging or self.csv_file:
+            return  # Already set up or logging disabled
+            
+        try:
+            timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+            filename = f"ChordsPy_{timestamp}.csv"
+            self.csv_file = open(filename, 'w', newline='')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)